modules/gapi/test/common/gapi_video_tests_common.hpp

```diff
@@ -0,0 +1,239 @@
+};
+
+namespace
+{
+inline void initTrackingPointsArray(std::vector<cv::Point2f>& points, size_t width, size_t height,
+                             size_t nPointsX, size_t nPointsY)
+{
+    if (nPointsX > width || nPointsY > height)
+    {
+        FAIL() << "Specified points number is too big";
+    }
+
+    size_t stepX = width  / nPointsX;
+    size_t stepY = height / nPointsY;
+
+
+    points.clear();
+    points.reserve(nPointsX * nPointsY);
+
+    for (size_t x = stepX / 2; x < width; x += stepX)
+    {
+        for (size_t y = stepY / 2; y < height; y += stepY)
+        {
+            Point2f pt(static_cast<float>(x), static_cast<float>(y));
+            points.push_back(pt);
+        }
+    }
+}
+
+template<typename Type>
+struct OptFlowLKTestInput
+{
+    Type& prevData;
+    Type& nextData;
+    std::vector<cv::Point2f>& prevPoints;
+};
+
+struct OptFlowLKTestOutput
+{
+    std::vector<cv::Point2f> &nextPoints;
+    std::vector<uchar>       &statuses;
+    std::vector<float>       &errors;
+};
+
+struct OptFlowLKTestParams
+{
+    OptFlowLKTestParams(): fileNamePattern(""), format(1), channels(0), pointsNum{0, 0},
+                           winSize(0), maxLevel(3), minEigThreshold(1e-4), flags(0) { }
+
+    OptFlowLKTestParams(const std::string& namePat, int chans,
+                        const std::tuple<size_t, size_t>& ptsNum, int winSz,
+                        const cv::TermCriteria& crit, const cv::GCompileArgs& compArgs,
+                        int flgs = 0, int fmt = 1, int maxLvl = 3, double minEigThresh = 1e-4):
+
+                        fileNamePattern(namePat), format(fmt), channels(chans),
+                        pointsNum(ptsNum), winSize(winSz), maxLevel(maxLvl),
+                        criteria(crit), minEigThreshold(minEigThresh), compileArgs(compArgs),
+                        flags(flgs) { }
+
+    std::string fileNamePattern;
+    int format;
+    int channels;
+    std::tuple<size_t,size_t> pointsNum;
+    int winSize;
+    int maxLevel;
+    cv::TermCriteria criteria;
+    double minEigThreshold;
+    cv::GCompileArgs compileArgs;
+    int flags;
+};
+} // namespace
+} // namespace opencv_test
+
+namespace
+{
+inline std::ostream& operator<<(std::ostream& os, const opencv_test::CompareForOptFlow&)
+{
+    return os << "compare_vector_f";
+}
+inline std::ostream& operator<<(std::ostream& os, const cv::TermCriteria& criteria)
+{
+    os << "{";
+    switch (criteria.type) {
+    case cv::TermCriteria::COUNT:
+        os << "COUNT; ";
+        break;
+    case cv::TermCriteria::EPS:
+        os << "EPS; ";
+        break;
+    case cv::TermCriteria::COUNT | cv::TermCriteria::EPS:
+        os << "COUNT | EPS; ";
+        break;
+    default:
+        os << "TypeUndifined; ";
+        break;
+    };
+
+    return os << criteria.maxCount << "; " << criteria.epsilon <<"}";
+}
+} // namespace
+
+#ifdef HAVE_OPENCV_VIDEO
+
+#include <opencv2/video.hpp>
```

Comment:
Better to have all includes written at the top of the file (ok, with an extra ifdef)

Reply:
fixed